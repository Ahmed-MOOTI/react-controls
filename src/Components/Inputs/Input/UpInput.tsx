// Imports
import * as React from 'react'
import { InputStyled } from './styles'
import { InputBaseComponent } from '../_Common/BaseControl/BaseControl'
import { UpInputProps, Validation } from './'
import TypeStringControl from '../_Common/Validation/TypeStringControl'

// EXports
export default class UpInput extends InputBaseComponent<UpInputProps, any> {
    public static defaultProps = {
        showError: true
    };

    constructor(p, c) {
        super(p, c);
        this.state = {
            value: p.value
        }
        var _self = this;
        if (this.props.validation && this.props.validation.length > 0) {
            this.props.validation.map(function (value: Validation, index: number) {
                if (value && value.pattern) {
                    _self._validationManager.addControl(new TypeStringControl(value.pattern, value.errorMessage));
                }
            });
        }
    }

    onChange(event: any) {
        return event.target.value;
    }

    renderControl() {
        const {type, onChange, value, validation, hasError, ...others } = this.props;
        return (
            <InputStyled
                value={this.state.value}
                type={type || "text"}
                hasError={this.props.hasError || this.hasError()}
                showError={this.props.showError}
                onChange={this.handleChangeEvent}>
                {this.props.children}
            </InputStyled>
        );
    }
}
