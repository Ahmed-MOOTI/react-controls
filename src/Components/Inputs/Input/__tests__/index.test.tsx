// Imports
import * as React from 'react';
import UpInput from '../';
import * as chai from 'chai';
import * as chaiEnzyme from 'chai-enzyme';
import { shallow, mount, render } from 'enzyme';
import * as renderer from 'react-test-renderer'
import 'jest-styled-components'

chai.use(chaiEnzyme());

describe('UpInput', () => {
  it('should render without throwing an error', function() {
    var _render = render(<UpInput />) ;
    var _input = _render.find("input");
    expect(_input.length).toBe(1);
  });

  it('should have a parent .up-input-group', function() {
    var _render = render(<UpInput width="medium" />) ;
    var _input = _render.find("div.up-input-group").parent();
    expect(_input.length).toBe(1);
  });

  it('should have a width', function() {
    var _render = renderer.create(<UpInput width="medium" />).toJSON() ;
    //expect(_render).toMatchStyledComponentsSnapshot();
    expect(_render).toMatchSnapshot();
  });
});