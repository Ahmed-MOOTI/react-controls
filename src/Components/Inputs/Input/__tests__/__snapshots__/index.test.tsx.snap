// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpInput should have a width 1`] = `


<div>
  <div
    className="hemHHH"
  >
    <div
      className="up-input-group"
    >
      <input
        className="up-input"
        dir="auto"
        disabled={undefined}
        maxLength={undefined}
        onChange={[Function]}
        placeholder={undefined}
        readOnly={undefined}
        type="text"
        value={undefined}
      />
      
    </div>
  </div>
  Â 
</div>
`;
